File: src/main/java/org/digma/intellij/plugin/assets/model/outgoing/SetCategoriesDataMessage.java
package org.digma.intellij.plugin.assets.model.outgoing;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record SetCategoriesDataMessage(@JsonProperty(value = "type",required = true) String type,
                                       @JsonProperty(value = "action",required = true)  String action,
                                       @JsonProperty(value = "payload",required = true) JsonNode payload )  {
    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public SetCategoriesDataMessage {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/assets/model/SetCategoriesDataMessage.kt
package org.digma.intellij.plugin.ui.assets.model

import com.fasterxml.jackson.databind.JsonNode
import org.digma.intellij.plugin.ui.jcef.JCEFGlobalConstants
import org.digma.intellij.plugin.ui.jcef.model.ErrorPayload


data class SetCategoriesDataMessage(val payload: JsonNode,val error: ErrorPayload? = null) {
    val type = JCEFGlobalConstants.REQUEST_MESSAGE_TYPE
    val action = "ASSETS/SET_CATEGORIES_DATA"
}